### ISSUE TITLE
luarocks 3.1.1: invalid value (boolean) at index 1 in table for 'concat' in dir.lua 

### ISSUE THREAD

**Issue by daurnimator** [luarocks/luarocks#1009]
Reported by @spc476 on lua mailing list:
```
[spc]lucy:~>luarocks
/usr/local/bin/lua: /home/spc/.luarocks/share/lua/5.3/luarocks/core/dir.lua:21: invalid value (nil) at index 1 in table for 'concat'
stack traceback:
        [C]: in function 'table.concat'
        /home/spc/.luarocks/share/lua/5.3/luarocks/core/dir.lua:21: in function 'luarocks.core.dir.path'
        /home/spc/.luarocks/share/lua/5.3/luarocks/cmd.lua:171: in upvalue 'get_lua_version'
        /home/spc/.luarocks/share/lua/5.3/luarocks/cmd.lua:291: in function 'luarocks.cmd.run_command'
        ...cks/lib/luarocks/rocks-5.3/luarocks/3.1.1-1/bin/luarocks:36: in main chunk
        [C]: in ?
```

Related to #1004

The refactor in #1006 was written assuming that `cfg.sysconfdir` was always available.
However, `get_lua_version()` is called before `cfg.init`: and hence `cfg.sysconfdir` hasn't been set up yet.
Is this intentional?

**Comment by ViperZer0** [comment]
I am also having this bug when trying to run `luarocks install`. Running just `luarocks` on its own works and prints out the help screen.

**Comment by daurnimator** [comment]
Should be fixed with 923f37301736155e1db07e540bc0e57e211d5380
